protocol/messenger_status_updates.go

```diff
@@ -0,0 +1,247 @@
+		ResendAutomatically: true,
+	}
+
+	_, err = m.sender.SendPublic(context.Background(), contactCodeTopic, rawMessage)
+	if err != nil {
+		return err
+	}
+
+	joinedCommunities, err := m.communitiesManager.Joined()
+	if err != nil {
+		return err
+	}
+	for _, community := range joinedCommunities {
+		rawMessage.LocalChatID = community.StatusUpdatesChannelID()
+		_, err = m.sender.SendPublic(context.Background(), rawMessage.LocalChatID, rawMessage)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Messenger) sendCurrentUserStatus() {
+	err := m.persistence.CleanOlderStatusUpdates()
+	if err != nil {
+		m.logger.Debug("Error cleaning status updates", zap.Error(err))
+		return
+	}
+
+	shouldBroadcastUserStatus, err := m.settings.ShouldBroadcastUserStatus()
+	if err != nil {
+		m.logger.Debug("Error while getting status broadcast setting", zap.Error(err))
+		return
+	}
+
+	if !shouldBroadcastUserStatus {
+		m.logger.Debug("user status should not be broadcasted")
+		return
+	}
+
+	currStatus, err := m.GetCurrentUserStatus()
+	if err != nil {
+		m.logger.Debug("Error obtaining latest status", zap.Error(err))
+		return
+	}
+
+	if err := m.sendUserStatus(*currStatus); err != nil {
+		m.logger.Debug("Error when sending the latest user status", zap.Error(err))
+	}
+}
+
+func (m *Messenger) sendCurrentUserStatusToCommunity(community *communities.Community) error {
+	shouldBroadcastUserStatus, err := m.settings.ShouldBroadcastUserStatus()
+	if err != nil {
+		return err
+	}
+
+	if !shouldBroadcastUserStatus {
+		m.logger.Debug("user status should not be broadcasted")
+		return nil
+	}
+
+	status, err := m.GetCurrentUserStatus()
+	if err != nil {
+		m.logger.Debug("Error obtaining latest status", zap.Error(err))
+		return err
+	}
+
+	status.Clock = uint64(time.Now().Unix())
+
+	err = m.settings.SaveSetting("current-user-status", status)
+	if err != nil {
+		return err
+	}
+
+	statusUpdate := &protobuf.StatusUpdate{
+		Clock:      status.Clock,
+		StatusType: protobuf.StatusUpdate_StatusType(status.StatusType),
+		CustomText: status.CustomText,
+	}
+
+	encodedMessage, err := proto.Marshal(statusUpdate)
+	if err != nil {
+		return err
+	}
+
+	rawMessage := common.RawMessage{
+		LocalChatID:         community.StatusUpdatesChannelID(),
+		Payload:             encodedMessage,
+		MessageType:         protobuf.ApplicationMetadataMessage_STATUS_UPDATE,
+		ResendAutomatically: true,
+	}
+
+	_, err = m.sender.SendPublic(context.Background(), rawMessage.LocalChatID, rawMessage)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *Messenger) broadcastLatestUserStatus() {
+	m.logger.Debug("broadcasting user status")
+	m.sendCurrentUserStatus()
+	go func() {
+		for {
+			select {
+			case <-time.After(5 * time.Minute):
+				m.sendCurrentUserStatus()
+			case <-m.quit:
+				return
+			}
+		}
+	}()
+}
+
+func (m *Messenger) SetUserStatus(newStatus int, newCustomText string) error {
```

Comment:
Don't you want to also save it in this case?

Reply:
It will be done inside `m.sendUserStatus()` in line 209